### ISSUE TITLE
Bad Design (or what) - the sleeping thread reuse phenomenon

### ISSUE THREAD

**Issue by KrishnaPG** [ipkn/crow#138]
With simple sleep code as below:
````
       crow::SimpleApp app;
	CROW_ROUTE(app, "/sleep/<int>")([](int n) {
		boost::this_thread::sleep(boost::posix_time::milliseconds(n * 1000));
		return "Done Sleep";
	});
		app.port(9000).multithreaded().run();
````
try calling the above from JS 20 times in a row and you can see that only small fixed no. of requests will be concurrently processed (based on your number of cores), and the remaining requests pending - while the threads are "sleeping".

In other words, the design is not taking advantage of the 'sleep' state of the thread to `schedule` the pending requests onto the same thread. Instead, the thread is waiting for the sleep to complete, before moving onto the next requests. i.e. threads are "busy", whether you are working are sleeping - not an elegant way to build servers.

This would mean, if you need to serve 10k concurrent requests, you would need to start 10,000 threads - a very very old pre c10k era design.

**Comment by ipkn** [comment]
No, it's wrong. You only need 10000 thread if you want 10000 synced sleeps. 

If you want to use handler in async manner, check `example_chat` which stores `res` and calls `res.end()` when message is ready.
